### PR TITLE
Fix attempting to report a chat message from multiplayer failing

### LINKED ISSUE #23477: Report popover in chat overlay display very awkwardly
### Type

Cosmetic

### Bug description

https://github.com/ppy/osu/pull/21736#issuecomment-1536507309

### Screenshots or videos

https://github.com/ppy/osu/assets/74862110/f18d3a8d-5722-4de8-9881-323859bd4a50

### Version

2023.511

### Logs

N/A

### PR DESCRIPTION
**Pull request by peppy**
..by adding a global `PopoverContainer`. It's still recommended to use local ones where it can work for now. In this case, making a local one would cause the popover to not have enough space to display.

Addresses https://github.com/ppy/osu/discussions/23684.
Closes #23477.

### CODE REVIEW

File: osu.Game/Tests/Visual/OsuManualInputManagerTestScene.cs [hunk 1]
```diff
@@ -50,9 +51,13 @@ protected OsuManualInputManagerTestScene()
...
                 var cursorDisplay = new GlobalCursorDisplay { RelativeSizeAxes = Axes.Both };
 
-                cursorDisplay.Add(content = new OsuTooltipContainer(cursorDisplay.MenuCursor)
+                cursorDisplay.Add(new PopoverContainer
                 {
                     RelativeSizeAxes = Axes.Both,
+                    Child = content = new OsuTooltipContainer(cursorDisplay.MenuCursor)
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                    },
```
Comment (peppy) [inline osu.Game/Tests/Visual/OsuManualInputManagerTestScene.cs:60]: This change is intentionally grouped away from the regressing test provided. It aims to match the `OsuGameBase` usage in a local way we can actually access from the test (to check the popover did display).

**Comment by peppy** [comment]
Hmm, it could be used to fix that problem, sure. I'll make the relevant change.

**Comment by bdach** [comment]
Doing this will regress https://github.com/ppy/osu/issues/14551, albeit only in the one case that is being fixed here (attempting to report from within multiplayer). Also tests are failing.

If I'm quite honest I would rather just delete the report option from the in-game chat. Reporting works in the lobby, and as things are it is basically impossible to report from the in-game chat due to bad UX.

**Comment by peppy** [comment]
I was making this change under the assumption that it would not have worked in the lobby.

What are your thoughts on https://github.com/ppy/osu/issues/23477? Should that have a local fix or something else?

**Comment by bdach** [comment]
> What are your thoughts on #23477? Should that have a local fix or something else?

I would probably try to see if it is possible to keep the popover local to the chat overlay but keep it the size of the entire screen. Something like

```
ChatOverlay (Height = 1)
    PopoverContainer (Height = 1)
        actual contents of chat
```

Falling that, I'd probably try to see how painful it would be to salvage the global container idea without breaking toasts.

All of the above is basically untested and wishful thinking, though.

**Comment by frenzibyte** [comment]
Having a global popover container is inevitable if we want to utilise the functionality of popovers, but it'll require further work like resolving https://github.com/ppy/osu-framework/issues/5818 for correct overlay colour and dependencies, and also proper placement of the global popover container to not appear over game-wide on-screen elements (i.e. resolving https://github.com/ppy/osu-framework/issues/5818).

Disabling reporting in game chat seems fine for the time being, it's kind of a rare use case and anyone who wants to report can very well do it from the lobby chat as mentioned above.

**Comment by bdach** [comment]
Given that the other PR disabling the problematic option was merged, can we close this now?

**Comment by peppy** [comment]
Sure let's close for the time being. I still feel like we'll want a global popup container in the future though.

**Pull request closed without merging**